spec/lib/jit_preloader/preloader_spec.rb

```diff
@@ -479,6 +480,41 @@
         end
       end
     end
+
+    context "with dive limit set" do
+      let!(:contact_book_1) { ContactBook.create(name: "The Yellow Pages") }
+      let!(:contact_book_2) { ContactBook.create(name: "The Yellow Pages") }
+      let!(:contact_book_3) { ContactBook.create(name: "The Yellow Pages") }
+      let!(:company1) { Company.create(name: "Company1", contact_book: contact_book_1) }
+      let!(:company2) { Company.create(name: "Company2", contact_book: contact_book_1) }
+      let!(:company3) { Company.create(name: "Company2", contact_book: contact_book_2) }
+      let!(:company4) { Company.create(name: "Company2", contact_book: contact_book_3) }
+      let!(:company5) { Company.create(name: "Company2", contact_book: contact_book_3) }
+
+      before do
+        JitPreloader.query_dive_limit = 3
```

Comment:
This is a global var, and if we ever enable test randomization (might even be enabled now) we run the risk of transients. We should reset after we're done. (Which means we need to ensure we have a way to remove the limit). 

Is this something we'll set globally for clio or per area?  Do we need a block form to set a limit for a chunk of code?

Reply:
I was initially considering a global value since this is based on the dive limit from MySql, which should not change from block to block of codes. 
But I can dig into it if a scenario supports that. Or concerns for a global setting?

Reply:
Update with the ability to roll out each aggregation method with dive limit.